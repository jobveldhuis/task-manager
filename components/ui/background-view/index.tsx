import { ReactNode } from "react";
import { LinearGradient } from "expo-linear-gradient";
import { StyleSheet } from "react-native";
import Colors from "../../../constants/Colors";

type BackgroundViewProps = {
  children: ReactNode;
};

export function BackgroundView({ children }: BackgroundViewProps): JSX.Element {
  return (
    <LinearGradient
      style={styles.container}
      colors={[
        Colors.backgroundGradient.primary,
        Colors.backgroundGradient.secondary,
      ]}
      testID="background"
    >
      {children}
    </LinearGradient>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "flex-start",
    justifyContent: "flex-start",
    paddingHorizontal: 25,
    paddingVertical: 150,
  },
});
