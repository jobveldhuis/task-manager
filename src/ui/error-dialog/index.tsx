export { ErrorDialog } from "./error-dialog.component";
