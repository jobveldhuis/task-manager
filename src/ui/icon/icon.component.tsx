import ExpoIcon from "@expo/vector-icons/FontAwesome5";
import { StyleSheet } from "react-native";
import { COLORS } from "@/util/colors.const";

export type IconName =
  | "calendar-alt"
  | "magic"
  | "check"
  | "user-check"
  | "check-double"
  | "flag-checkered"
  | "list"
  | "edit"
  | "long-arrow-alt-right"
  | "exclamation-triangle";

type IconProps = {
  name: IconName;
  variant?: (typeof variants)[number];
};

const variants = ["primary", "inverse"] as const;

export function Icon({ name, variant = "primary" }: IconProps): JSX.Element {
  return <ExpoIcon name={name} style={[styles.base, styles[variant]]} />;
}

const styles = StyleSheet.create({
  base: {
    fontSize: 16,
  },
  primary: {
    color: COLORS.primary,
  },
  inverse: {
    color: COLORS.inverse,
  },
});
