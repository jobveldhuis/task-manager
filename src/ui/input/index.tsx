export { Input } from "./input.component";
export { DateInput } from "./date-input.component";
export { MultilineInput } from "./multiline-input.component";
