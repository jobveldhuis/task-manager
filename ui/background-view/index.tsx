import { ReactNode } from "react";
import { LinearGradient } from "expo-linear-gradient";
import { StyleSheet } from "react-native";
import { COLORS } from "@/util/colors.const";

type BackgroundViewProps = {
  children: ReactNode;
};

export function BackgroundView({ children }: BackgroundViewProps): JSX.Element {
  return (
    <LinearGradient
      style={styles.container}
      colors={[
        COLORS.backgroundGradient.primary,
        COLORS.backgroundGradient.secondary,
      ]}
      testID="background"
    >
      {children}
    </LinearGradient>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "flex-start",
    justifyContent: "flex-start",
    paddingHorizontal: 32,
    paddingVertical: 120,
  },
});
