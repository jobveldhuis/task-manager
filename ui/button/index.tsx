import {
  GestureResponderEvent,
  Pressable,
  StyleSheet,
  View,
  ViewStyle,
} from "react-native";
import { COLORS } from "@/util/colors.const";
import { Text } from "../text";

type ButtonProps = {
  title: string;
  onPress?: (event: GestureResponderEvent) => void;
  style?: ViewStyle;
};

export function Button({
  title,
  style,
  onPress: handlePress,
}: ButtonProps): JSX.Element {
  return (
    <Pressable
      style={({ pressed: isPressed }) => [
        styles.container,
        { opacity: isPressed ? 0.6 : 1 },
      ]}
      onPress={handlePress}
    >
      <View style={[styles.button, style]}>
        <Text variant="button">{title}</Text>
      </View>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    width: "100%",
  },
  button: {
    backgroundColor: COLORS.button.primary,
    padding: 10,
    borderRadius: 20,
    width: "100%",
    textAlign: "center",
  },
});
