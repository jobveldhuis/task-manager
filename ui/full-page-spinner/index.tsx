export { FullPageSpinner } from "./full-page-spinner.component";
