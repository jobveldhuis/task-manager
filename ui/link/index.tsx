import { GestureResponderEvent, Pressable, StyleSheet } from "react-native";
import { Text } from "@/ui/text";
import { Icon } from "@/ui/icon";

type LinkProps = {
  text: string;
  onPress: (event: GestureResponderEvent) => void;
  shouldHideArrow?: boolean;
};

export function Link({
  shouldHideArrow = false,
  text,
  onPress: handlePress,
}: LinkProps): JSX.Element {
  return (
    <Pressable style={styles.container} onPress={handlePress}>
      <Text variant="link">{text}</Text>
      {!shouldHideArrow && <Icon name="long-arrow-alt-right" />}
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    gap: 8,
  },
});
