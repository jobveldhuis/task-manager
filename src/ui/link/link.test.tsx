import { render, screen } from "@testing-library/react-native";
import { Icon } from "@/ui/icon";
import { Link } from "./link.component";

jest.mock("@expo/vector-icons/FontAwesome5");

describe("Link", () => {
  const handlePress = jest.fn();

  const component = <Link text="Foo" onPress={handlePress} />;

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("should match snapshot", () => {
    expect(render(component).toJSON()).toMatchSnapshot();
  });

  it("should display the text passed as a prop", () => {
    render(component);
    expect(screen.getByText("Foo")).toBeTruthy();
  });

  it("should display an icon by default", () => {
    const { UNSAFE_getByType: getByType } = render(component);
    const icon = getByType(Icon);

    expect(icon).toBeTruthy();
  });

  it("should hide the icon if passed as prop", () => {
    const { UNSAFE_queryByType: queryByType } = render(
      <Link text="Foo" onPress={handlePress} shouldHideArrow />,
    );

    const icon = queryByType(Icon);
    expect(icon).toBeFalsy();
  });
});
