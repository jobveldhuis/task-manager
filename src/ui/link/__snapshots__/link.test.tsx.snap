// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
      },
      {
        "opacity": 1,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000C49",
          "fontSize": 16,
          "fontWeight": "600",
        },
        {
          "color": "#000C49",
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Foo
  </Text>
</View>
`;
